### DISCUSSION TITLE
Export logs to external (S3) Storage as Retention

### DISCUSSION THREAD

**Discussion by stefandevo** [datalust/seq-tickets#1611] [Help] [answered]
As far as I can see, retention policy will just Delete the events.
However, wouldn't it be a great feature that you can export through json exporter to cloud storage (s3, gcs, ...).
This would +export your old logs and +delete the logs locally.

I understand this is possible in some extend now:
- use the json archive app and export it daily to json files on local disk
- create an app to push these json files daily to s3 storage (don't think there is an app for)
- create an app to delete these json files if there are push (don't think there is an app for) - or combine it in previous step.
- set retention period

Any guidance if this would be an option to implement?

**Comment by nblumhardt** [comment, marked as answer]
Hi Stefan,

Thanks for you message. RE syncing JSON archive files to S3, it appears that a scheduled task/cron job running the AWS CLI's `sync` command would achieve this:

https://docs.aws.amazon.com/cli/latest/reference/s3/sync.html

The command would be something like:

```powershell
aws s3 sync "C:\Archive" s3://logs-archive
```

Basing the scheduled task that also deletes (local) copies of the files past a certain age should cover the remaining item. In PowerShell, according to [this Stack Overflow answer](https://stackoverflow.com/a/40608919/138206), it should be something like:

```powershell
Get-ChildItem "C:\Archive" -Recurse -File | Where CreationTime -lt  (Get-Date).AddDays(-15)  | Remove-Item -Force
```

That's probably the lowest-friction option I can think of, but it's also possible to write Seq apps in C#, or run any arbitrary executable or script as an app (Seq just sends events one-per-line over STDIN).

Let me know if this helps!

**Reply by stefandevo** [reply]
Thanks for these suggestions. Will try to look into Seq apps to see if I can come up with something. Hoped I could look at the JsonArchive but that's not in C# :-)

**Reply by stefandevo** [reply]
Just looked at the Seq apps. Basically I just need to use https://github.com/serilog-contrib/Serilog.Sinks.AmazonS3 Serilog sink and just push every event on that. Correct?

**Reply by nblumhardt** [reply]
Hi @stefandevo - that sounds like a reasonable approach, let me know if you run into any challenges!

Best regards,
Nick
